src/js/libcs/Evaluator.js

```diff
@@ -151,3 +154,29 @@ function analyse(code) {
         namespace.create(name);
     return res;
 }
+
+var callStack = [];
+
+function labelCode(code, label) {
+    function run() {
+        return evaluate(code);
+    }
+    return {
+        ctype: "infix",
+        args: [],
+        impl: function() {
+            callStack = [{
+                oper: label
+            }];
+            var res = evaluate(code);
+            callStack = [];
+            return res;
+        }
+    };
+}
+
+function printStackTrace(msg) {
+    csconsole.err(msg + "\n" + callStack.map(function(frame) {
+        return "  at " + frame.oper;
```

Comment:
Line numbers with respect to the script should be doable if we stop [discarding](https://github.com/CindyJS/CindyJS/pull/354) the [position information we have](https://github.com/CindyJS/CindyJS/blob/v0.8.3/src/js/libcs/Parser.js#L308-L314). But as you're usually looking at the HTML file as a whole, those line numbers would have to be offset by the start of the `<script>` tag. I doubt there is an easy [way to obtain that](http://stackoverflow.com/q/16298945/1468366). Perhaps it would be better to print some context around the point of invocation. But as that would require some more work, I'd leave that for a later improvement.